internal/service/base/resource_agreement_localization_revision.go

```diff
@@ -476,17 +474,36 @@ func (r *AgreementLocalizationRevisionResource) ImportState(ctx context.Context,
 func (p *AgreementLocalizationRevisionResourceModel) expand() (*management.AgreementLanguageRevision, diag.Diagnostics) {
 	var diags, d diag.Diagnostics
 
+	// Provide a buffer for the effective_at time to allow for processing in Terraform
+	//     when setting the value in the plan if not provided by the user
+	// Grace period used to prevent the effective_at time from being set in the past by the user in HCL
+	now := time.Now().UTC().Truncate(time.Second)
+	buffer := 1 * time.Minute
+	grace := 1 * time.Second
+	// Flag for notification to indicate if a generated effective_at time was used
+	usedGenerated := false
+
 	var t time.Time
 
 	if !p.EffectiveAt.IsNull() && !p.EffectiveAt.IsUnknown() {
 		t, d = p.EffectiveAt.ValueRFC3339Time()
 		diags.Append(d...)
```

Comment:
There's no logic to return early if this generates an error now

Reply:
I think commit this afternoon addresses this